packages/@expo/metro-config/src/transform-worker/metro-transform-worker.ts

```diff
@@ -700,7 +700,12 @@ function getBabelTransformArgs(
     options: {
       ...babelTransformerOptions,
       enableBabelRCLookup: config.enableBabelRCLookup,
-      enableBabelRuntime: config.enableBabelRuntime,
+      // NOTE(@kitten): This shouldn't be relevant via this code path. However, in case it does,
+      // this prevents us from adding imports/requires to @babel/runtime when we're transforming a script
+      enableBabelRuntime:
+        options.type === 'script' && config.enableBabelRuntime !== false
+          ? false
+          : config.enableBabelRuntime,
```

Comment:
```suggestion
        options.type === 'script' ? false : config.enableBabelRuntime,
```

nit: lean toward to keep it easier - to disable babel runtime anyway for "script", since polyfills cannot contain babel runtime "require" at all.
though it has redundant writes (when platformOptions.enableBabelRuntime is false, we set it to false again), i feel the code might be clear.